ConsoleApp6/PaymentServices.cs

```diff
@@ -15,35 +18,53 @@ public PaymentServices(Products p)
             this.UPC = p.UPC;
             this.Tax = Products.Tax;
             this.Discount = Products.Discount;
+            this.specialUPC = Products.specialUPC;
+            this.upcDiscount = Products.UPC_Discount;
          
```

Comment:
regarding the Product class, the discount and tax rate are not part of the Product's real-world representation, not part of it, so we can look for a better way to separate the price calculations data from the product itself